File: app/src/main/java/metri/amit/sampleapp/view/SplashActivity.java
package metri.amit.sampleapp.view;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * Created by amitmetri on 05,April,2021
 * Splash activity has been added to improve app launch experience.
 * Splash activity will render the UI directly from the theme
 * so that there is not time gap before rendering layout and no blank white screen appearance.
 */
public class SplashActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        /*
        * Navigate to MainActivity post 1 second delay
        * */
        new Handler().postDelayed(() -> {
            startActivity(new Intent(SplashActivity.this, MainActivity.class));
        }, 1000);
    }
}

File: app/src/main/java/metri/amit/sampleapp/view/SplashActivity.kt
package metri.amit.sampleapp.view

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity

/**
 * Created by amitmetri on 28,April,2021
 */
class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        /*
        * Navigate to MainActivity post 1 second delay
        * */
        Handler(Looper.getMainLooper())
                .postDelayed({ startActivity(Intent(this@SplashActivity,
                        MainActivity::class.java))
                             finish()}, 1000)
    }
}